diff --git a/test/support/apps/phx1_6/config/config.exs b/test/support/apps/phx1_6/config/config.exs
index 3c2eed0..df7ce7c 100644
--- a/test/support/apps/phx1_6/config/config.exs
+++ b/test/support/apps/phx1_6/config/config.exs
@@ -47,6 +47,10 @@ config :logger, :console,
 # Use Jason for JSON parsing in Phoenix
 config :phoenix, :json_library, Jason
 
+config :torch,
+  otp_app: :phx1_6,
+  template_format: "eex" || "slime"
+
 # Import environment specific config. This must remain at the bottom
 # of this file so it overrides the configuration defined above.
 import_config "#{config_env()}.exs"
diff --git a/test/support/apps/phx1_6/lib/phx1_6_web/endpoint.ex b/test/support/apps/phx1_6/lib/phx1_6_web/endpoint.ex
index f42c175..594a955 100644
--- a/test/support/apps/phx1_6/lib/phx1_6_web/endpoint.ex
+++ b/test/support/apps/phx1_6/lib/phx1_6_web/endpoint.ex
@@ -22,6 +22,14 @@ defmodule Phx16Web.Endpoint do
     gzip: false,
     only: ~w(assets fonts images favicon.ico robots.txt)
 
+  plug(
+    Plug.Static,
+    at: "/torch",
+    from: {:torch, "priv/static"},
+    gzip: true,
+    cache_control_for_etags: "public, max-age=86400"
+  )
+
   # Code reloading can be explicitly enabled under the
   # :code_reloader configuration of your endpoint.
   if code_reloading? do
diff --git a/test/support/apps/phx1_6/mix.exs b/test/support/apps/phx1_6/mix.exs
index 6cd9404..f24f00c 100644
--- a/test/support/apps/phx1_6/mix.exs
+++ b/test/support/apps/phx1_6/mix.exs
@@ -48,7 +48,8 @@ defmodule Phx16.MixProject do
       {:telemetry_poller, "~> 1.0"},
       {:gettext, "~> 0.18"},
       {:jason, "~> 1.2"},
-      {:plug_cowboy, "~> 2.5"}
+      {:plug_cowboy, "~> 2.5"},
+      {:torch, path: "../../../../"}
     ]
   end
 
